include/mata/nfa/delta.hh

```diff
@@ -143,9 +162,12 @@ public:
      * @return Number of states in the whole Delta, including both source and target states.
      */
     size_t num_of_states() const { return posts.size(); }
+    //should it just be size() ?
```

Comment:
but `num_of_transitions()` is still not implemented, right?

Reply:
It is implemented already.

Reply:
`size` is fine, however, in the context of `Delta` it is not clear what is meant by that. Is it used somewhere? (except `Nfa::size()`)

Reply:
No, it is not used anywhere except for a few tests. We use num_of_states() and num_of_transitions() everywhere else instead.

Reply:
so stay with num_of_transitions/states? Or post_size instead of num_of_transitions?

Reply:
I would highly encourage us to use `num_of_something()` everywhere where `size()` is ambiguous. `post_size()` is ambiguous again because of renaming of both `Post` and `Move` to `...Post`. But it could be used for `Delta` if what we want to describe is the structural quality of the `Delta` class instead of the semantic property of `Delta` as a transition relation of an NFA.

Reply:
I would be consistent. Why `Delta::num_of_states` and `Nfa::size`?

Reply:
Yes that is weird, and it should be more consistent. Nfa::size is in fact bad, one never knows what it is.  Nfa::num_of_states sounds good. But initial and final can still have size(), right? since they are just sets.

Reply:
`Nfa::size()` is the number of transitions, and we agreed recently that everything to do with transitions should be in `Delta`. Therefore, `Nfa::size()` should be removed, or modified to return the number of states in the automaton. But if we have `Delta::num_of_states()` and `Delta::num_of_transitions()`, we might want to remove `Nfa::size()` altogether and only keep `Nfa::num_of_states()` to be explicit and consistent. No more wondering what `Nfa::size()` means.

Reply:
I agree with not using `size()` in `Nfa/Delta`. There probably should be `Nfa::num_of_states` and `Delta::num_of_transitions`, but I am not sure if we should keep `Delta::num_of_states`. Final/inital can just keep `size()`, they are sets, so it is obvious what it returns.

Reply:
Resolved. Using `num_of_states()` and `num_of_transitions()` everywhere. `Nfa::initial` and `Nfa::final` keep `size()`, as they are simple sets. Discussion about implementing wrapper around these sets will continue in #325.